### ISSUE TITLE
How to import on ESM project

### ISSUE THREAD

**Issue by jacargentina** [onderonur/react-infinite-scroll-hook#38]
I'm migrating my remix app to be all ESM, my code was like this

```
import useInfiniteScroll from 'react-infinite-scroll-hook';
```

But now it produces error 

```
TypeError: __vite_ssr_import_5__.default is not a function or its return value is not iterable
```

Stack pointing to my code:

<img width="474" alt="Captura de pantalla 2024-04-04 a la(s) 12 43 50" src="https://github.com/onderonur/react-infinite-scroll-hook/assets/517228/e2d81e96-3856-4779-ba24-3f5145a85ff8">

## I've tried

import { default as useInfiniteScroll } from 'react-infinite-scroll-hook';
import * as useInfiniteScroll from 'react-infinite-scroll-hook';

No luck!

**Comment by jacargentina** [comment]
https://github.com/jacargentina/my-remix-app.git

**Comment by jacargentina** [comment]
Maybe I've found the reason: this package is "client only", so when SSR is running the error is produced.

```tsx
import { useHydrated } from "remix-utils/use-hydrated";

export default function Index() {
  let isHydrated = useHydrated();

  const inf = isHydrated
    ? useInfiniteScroll({
        loading: true,
        hasNextPage: false,
        onLoadMore: () => {},
      })
    : useRef();
...
}
```

This "patch" makes error go away. 

Anyway, I would love a better way so I don't have to patch every place where I use `useInfiniteScroll` on my code?

**Comment by onderonur** [comment]
Hello @jacargentina 👋

Firstly, thanks for the findings you shared 🙏

I will publish an alpha version today or tomorrow and notify you from here. You can try it and if it solves the issue, I will share the solution for documentation purposes, you know if any other library maintainer comes across this by searching it on Google etc.

Good thing is, I was already planning migrating away from tsdx and it looks like the new structure will handle ESM better.

**Comment by jacargentina** [comment]
@onderonur great news!

I should correct myself on my previous "finding": the problem seems to be related on CJS/ESM handling on package.json: I've patched it like this and it works just fine now! The "exports" is key here.

```json
{
  "version": "4.1.1",
  "license": "MIT",
  "main": "dist/index.js",
  "type": "module",
  "exports": {
    ".": {
	"require": {
	    "default": "./dist/index.js"
	},
	"import": {
	    "default": "./dist/react-infinite-scroll-hook.esm.js"
	}
    }
  },
  "typings": "dist/index.d.ts",
  "files": [
    "dist",
    "src"
  ],
  "engines": {
    "node": ">=10"
  },
  "scripts": {
    "start": "tsdx watch",
    "build": "tsdx build",
    "test": "tsdx test --passWithNoTests",
    "lint": "tsdx lint --fix",
    "prepare": "tsdx build && husky install"
  },
  "peerDependencies": {
    "react": ">=16.8.0"
  },
  "name": "react-infinite-scroll-hook",
  "description": "A simple hook to create infinite scroll components",
  "author": "onderonur",
  "homepage": "https://onderonur.github.io/react-infinite-scroll-hook",
  "repository": {
    "type": "git",
    "url": "https://github.com/onderonur/react-infinite-scroll-hook.git"
  },
  "bugs": {
    "url": "https://github.com/onderonur/react-infinite-scroll-hook/issues"
  },
  "keywords": [
    "react",
    "react-hooks",
    "infinite-scroll",
    "react-component"
  ],
  "devDependencies": {
    "@types/react": "^17.0.3",
    "@types/react-dom": "^17.0.3",
    "@typescript-eslint/eslint-plugin": "^4.20.0",
    "@typescript-eslint/parser": "^4.20.0",
    "all-contributors-cli": "^6.20.0",
    "eslint": "^7.23.0",
    "eslint-config-airbnb": "^18.2.1",
    "eslint-config-prettier": "^7.2.0",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-jsx-a11y": "^6.4.1",
    "eslint-plugin-prettier": "^3.3.1",
    "eslint-plugin-react": "^7.23.1",
    "eslint-plugin-react-hooks": "^4.2.0",
    "husky": "^6.0.0",
    "prettier": "^2.2.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "tsdx": "^0.14.1",
    "tslib": "^2.1.0",
    "typescript": "^4.2.3"
  },
  "dependencies": {
    "react-intersection-observer-hook": "^2.1.1"
  }
}
```

**Comment by onderonur** [comment]
Hello again @jacargentina 👋 

Can you try `react-infinite-scroll-hook@5.0.0-alpha.0` please?

I tried it in the repository you shared and it looks like the issue is resolved.

After your confirmation, I will share a summary when possible and close this issue.

And after that, I will publish the new major version. Until then, you can use the `alpha` version.

**Comment by jacargentina** [comment]
@onderonur  It works!!!! 👏🏻

**Comment by onderonur** [comment]
Hi again @jacargentina 👋 

I've just published `v5.0.0` which includes the fix for this issue.

Thanks again 🙏
